pageserver/src/tenant/timeline.rs

```diff
@@ -3347,9 +3472,23 @@ impl Timeline {
         target_file_size: u64,
         ctx: &RequestContext,
     ) -> Result<CompactLevel0Phase1Result, CompactionError> {
+        let mut stats = CompactLevel0Phase1StatsBuilder {
+            version: Some(1),
+            tenant_id: Some(self.tenant_id),
+            timeline_id: Some(self.timeline_id),
+            ..Default::default()
+        };
+
+        let begin = tokio::time::Instant::now();
         let layers = self.layers.read().await;
+        let now = tokio::time::Instant::now();
+        stats.first_read_lock_acquisition_micros =
+            DurationRecorder::Recorded(RecordedDuration(now - begin), now);
```

Comment:
```suggestion
            DurationRecorder::Recorded(RecordedDuration(begin.elapsed()), now);
```

... so that it's more concise :)

Reply:
See https://github.com/neondatabase/neon/pull/4527#discussion_r1232280600